### PR TITLE
Julia shts - add adjoint synthesis direction

### PR DESCRIPTION
**Pull request by LeeoBianchi**

### CODE REVIEW

**Comment by mreineck** [comment]
Thanks a lot for this!

What's your opinion on the `*start` arrays? Should we honour Julia's 1-based convention and expect 1-based indices here? Or should we perhaps rename `*start` to `*offset`, where zero-based numbers would feel more natural again?

You mention at one point in the sources that the code segfaults; is this still the case? If so, I can have a closer look if you could provide me with a small example.

**Comment by LeeoBianchi** [comment]
Regarding the `*start` arrays, to me it looks like the smoother solution is to switch to the 1-base convention on the Julia side and implement the conversion to the 0-base in the wrapper file `ducc_julia.cc`. If we want to keep everything on a 0-base, the `offset` name solution sounds good tho.

Yes, at that line the code still segfaults. I'm emailing you now a fairly commented example.

Thank you very much.

File: julia/ducc0.jl [hunk 1]
```diff
@@ -302,23 +300,91 @@ function sht_alm2leg(
...
+  lmax::Unsigned,
+  mval::StridedArray{Csize_t,1},
+  mstart::StridedArray{Csize_t,1}, # FIXME: this is still 0-based at the moment
+  lstride::Int,
+  theta::StridedArray{Cdouble,1},
+  nthreads::Unsigned,
+) where {T}
+  ncomp = size(leg, 3)
+  nm = length(mval)
+  alm = Array{Complex{T}}(undef, (nm, ncomp))
```
Comment (mreineck) [inline julia/ducc0.jl]: I think this should be
```
alm = Array{Complex{T}}(undef, (mstart[size(mstart,1)]+lmax+1, ncomp))
```
but I might be off by 1. Just `nm` as size is too small.

Comment (mreineck) [reply julia/ducc0.jl]: Of course this is under the assumption that the `m` values are sorted. The general solution would be something like `max(mstart)+lmax+1`.

**Comment by mreineck** [comment]
Merging now, work will most likely continue in another dedicated repository.

**Pull request merged**